### ISSUE TITLE
Calendar timepicker stepMinute should show zero-based multiples of step

### ISSUE THREAD

**Issue by jamesmcglinn** [primefaces/primereact#2301]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**
Please fork the codesandbox below and create a case demonstrating your bug report. Issues without a codesandbox have much less possibility to be reviewed.

https://codesandbox.io/s/primereact-test-forked-lsctl

**Current behavior**
When using the Calendar timepicker's stepMinute prop the default value's minutes are the exact current time, regardless of whether the current time minutes are a multiple of stepMinute.

**Expected behavior**
The default value should have minutes set to a multiple of the stepMinute prop.

This has already been fixed in primefaces and primevue:
https://github.com/primefaces/primefaces/issues/6047
https://github.com/primefaces/primevue/issues/115

I'm happy to create a PR with the same changes if you'd like.

**Minimal reproduction of the problem with instructions**
https://codesandbox.io/s/primereact-test-forked-lsctl

To reproduce open the codesandbox when the current time isn't on the hour, or 20/40 minutes past the hour. At all other times it will default to the current time, and the increment/decrement minute buttons will add/subtract 20 minutes from the arbitrary current time.

**Please tell us about your environment:**
https://codesandbox.io/s/primereact-test-forked-lsctl

* **React version:**
17.0.2

* **PrimeReact version:**
6.5.1

* **Browser:**
All

* **Language:**
All

**Comment by mertsincan** [comment]
Hi @jamesmcglinn,

> I'm happy to create a PR with the same changes if you'd like.
I like it ;) Could you please create a PR? Thanks a lot @jamesmcglinn ;)

**Comment by melloware** [comment]
PR Submitted

**Comment by melloware** [comment]
@MNazakov it looks like its working to me?  It bounced you back and forth between 12:30 and 12:00 ?  That is expected for `stepMinute={30}` in my opinion unless I am missing something?

**Comment by MNazakov** [comment]
Hi @melloware, thanks for the reply, as far as I understand from the OP, the issue is that the default value is wrong ( when calendar is shown )

> **Current behavior**
> When using the Calendar timepicker's stepMinute prop the default value's minutes are the exact current time, regardless of whether the current time minutes are a multiple of stepMinute.

The problem is the initial value. For example if the current time is _15:53_ and stepMinute is set to _30_ the initial time shown in the date picker will be 15:53 and will change to 16:00 for up button ( or 15:30 if the user has pressed the down button ) _only_ after the interaction has happened. This means that the user has can submit the time with an _invalid_ value depending on the use-case.

Please let me know if I misunderstood the OP's issue description, or if what I'm saying is not related to it

Thanks!

**Comment by melloware** [comment]
ahh yes its not checking the initial minute.  I just opened a new ticket: https://github.com/primefaces/primereact/issues/3770

**Comment by melloware** [comment]
I fixed it!  It will be in 8.7.4
